pkg/packetbrokeragent/grpc_gspba.go

```
@@ -159,12 +160,25 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "contact_info") {
 		adminContact, techContact := toPBContactInfo(req.Gateway.GetContactInfo())
-		updateReq.AdministrativeContact = &packetbroker.ContactInfoValue{
-			Value: adminContact,
+
+		// Only replaces the administrative and technical contacts if no error happens.
```

What kind of errors would occur that can just be swallowed?

---

Since a gateway's admin/tech contact cannot be a user/org that doesn't exist there and we are using `WithClusterAuth`, there should not be an error in regards to business logic. With this in mind the reason for swallowing the error would be to not interrupt the `UpdateGateway` procedure due to error in the connection between the PBA and IS. Regardless I should at least log the error in case something unexpected happens. 

Let me know if you agree with my logic in this case

---

It's possible that the user is deleted after it is set as the contact, and in these cases it will be 'not found', while still being referenced in the contact fields.

As you have said, there are many reasons unrelated to the business logic itself that can cause this call to fail - connection issues, database issues, and others. While the caller cannot fix them, it should at least log them with a warning. You can consider skipping the logging for 'not found entities'.